src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -564,24 +644,90 @@ private void SetPatternLocalScopes(BoundObjectPattern pattern)
+            stack.Push(node.Right);
+
+            BoundExpression current = node.Left;
+            while (current is BoundBinaryOperator { InterpolatedStringHandlerData: null } expr)
+            {
+                stack.Push(expr.Right);
+                current = expr.Left;
+            }
+
+            visitExpression(current);
+            while (stack.Count > 0)
+            {
+                current = stack.Pop();
+                visitExpression(current);
+            }
+
+            stack.Free();
+            return null;
+
+            void visitExpression(BoundExpression node)
+            {
+                if (node is BoundBinaryOperator { InterpolatedStringHandlerData: { } interpolationData } expr)
+                {
+                    visitBinaryInterpolation(expr, interpolationData);
+                }
+                else
+                {
+                    Visit(node);
+                }
+            }
+
+            void visitBinaryInterpolation(BoundBinaryOperator node, InterpolatedStringHandlerData interpolationData)
+            {
+                var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, uint)>.GetInstance();
+                VisitInterpolatedStringHandlerDataAndGetReceiverPlaceholder(placeholders, interpolationData);
+
+                using var _ = new PlaceholderRegion(this, placeholders);
+                Visit(node.Left);
+                Visit(node.Right);
+            }
+        }
+
+        private void VisitArgumentsAndGetArgumentPlaceholders(BoundExpression? receiverOpt, ImmutableArray<BoundExpression> arguments, uint receiverEscapeScope = uint.MaxValue)
+        {
+            Visit(receiverOpt);
+
+            for (int i = 0; i < arguments.Length; i++)
             {
+                var arg = arguments[i];
                 if (arg is BoundConversion { ConversionKind: ConversionKind.InterpolatedStringHandler, Operand: BoundInterpolatedString or BoundBinaryOperator } conversion)
                 {
                     var interpolationData = conversion.Operand.GetInterpolatedStringHandlerData();
-                    GetInterpolatedStringPlaceholders(placeholders, interpolationData, receiverOpt, arguments);
+                    var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, uint)>.GetInstance();
+                    GetInterpolatedStringPlaceholders(placeholders, interpolationData, receiverOpt, i, arguments, receiverEscapeScope);
+                    _ = new PlaceholderRegion(this, placeholders);
                 }
+                Visit(arg);
             }
-            return new PlaceholderRegion(this, placeholders);
         }
 
         public override BoundNode? VisitCall(BoundCall node)
         {
-            using var _ = GetArgumentPlaceholders(node.ReceiverOpt, node.Arguments);
-            base.VisitCall(node);
+            VisitArgumentsAndGetArgumentPlaceholders(node.ReceiverOpt, node.Arguments);
```

Comment:
>VisitArgumentsAndGetArgumentPlaceholders [](http://example.com/codeflow?start=12&length=40)

The original bug was here. Preceding arguments should be visited before evaluating the escape scope of subsequent placeholders.